### ISSUE TITLE
Harbor HA design for production

### ISSUE THREAD

**Issue by shahidv3** [goharbor/harbor#14286]
Hi Team
do you have any guidelines on harbor HA design for prod environment without using Helm charts.

many thanks
md

**Comment by phin1x** [comment]
@shahidv3 how do you want to deploy harbor, multiple server with docker?

**Comment by shahidv3** [comment]
hi @phin1x , Yes I want to deploy harbor, multiserver with docker.

thanks

**Comment by phin1x** [comment]
for multiserver you have to setup a dedicated redis and postgres cluster and configure those clusters in the external_database and external_redis section in the harbor config file. the directories /data/secrets and /data/registry (if you use local filesystem storage for the registry) must be shared accross all servers, for example with nfs or sync the /data/secrets file manually and use s3 storage for the registry (minio for example). then just call the install script on all machines and loadbalance the http/s ports with a external loadbalance.

to sum up, there is no easy, out of the box way of running harbor on multiple docker server.
